Update collection editing and sharing UI

emma-sg opened the pull request:
Closes #3130

## Changes

- adds inline collection name and caption editing on the detail page
- changes "copy link" button text from "Copy Link" to "Copy Public Link" for clarity when collection is public
- modifies the collection detail page to make the access icon next to the collection title into a clickable button that opens the sharing tab in the edit dialog 
- adds a dedicated share button to the header actions with a share icon that also opens the sharing tab
- normalizes toast ids to "update" to not clutter screen when making consecutive changes
- fixes a variety of wrapping/clipping/text truncation issues on the collection detail page, public collection page, and collection list view
- adds a minimum collection name length that's validated on the back-end, as previously the name could be set to an empty string, which would cause requests to fail involving that collection
- fixes a few bugs with smaller viewport sizes

## Demo


https://github.com/user-attachments/assets/ba7a78ed-b603-4830-a708-74f8c1e60575

emma-sg committed:
[wip] update collection sharing UI and copy button

- updates the share collection component to add "name" and "size"
attributes to the copy button for improved accessibility and styling -
changes button text from "Copy Link" to "Copy Public Link" for clarity -
modifies the collection detail page to make the access icon next to the
collection title a clickable button that opens the sharing tab in the
edit dialog - adds a dedicated share button to the header actions with a
share icon that also opens the sharing tab

emma-sg committed:
[wip] update collection sharing UI and copy button

- updates the share collection component to add "name" and "size"
attributes to the copy button for improved accessibility and styling -
changes button text from "Copy Link" to "Copy Public Link" for clarity -
modifies the collection detail page to make the access icon next to the
collection title a clickable button that opens the sharing tab in the
edit dialog - adds a dedicated share button to the header actions with a
share icon that also opens the sharing tab

emma-sg committed:
wip inline collection name/caption editing

emma-sg committed:
improve editable text field and collection header

- remove asynchronous text measurement for better performance
- add validation UI with character counter for editable fields
- fix layout issues with long collection names and captions
- improve responsive design with better truncation and wrapping
- fix page layout padding consistency

emma-sg committed:
remove extra logging

emma-sg committed:
various fixes

- remove event listeners on disconnected callback
- remove unnecessary initialization
- fix some translated strings
- use same id for generic update toasts
- remove empty nested icon component

emma-sg committed:
fix collection list max width on titles

emma-sg committed:
improve text field interactions

- improve hitboxes
- show warning about unsaved changes when failing validation
- improve validation warning display

emma-sg committed:
make access icons into buttons for crawlers

emma-sg committed:
throw error when text measurement returns nothing (should never happen,
but have noticed it when reloading sometimes)

emma-sg committed:
fix a few more little details

emma-sg committed:
ensure tab is correctly set when re-opening collection edit dialog

emma-sg committed:
remove redundant "edit collection settings" icon in toolbar

emma-sg committed:
Revert "remove redundant "edit collection settings" icon in toolbar"

This reverts commit 27fe0653c6c819fa5555f1362f637633b9dfb8ef.

emma-sg committed:
remove *correct* redundant edit button

emma-sg committed:
fix alignment

emma-sg committed:
fix annotation in backend

emma-sg committed:
fix small layout bugs with public page, header, and public collection
page

tw4l reviewed (commented):
Nice! Overall working very well.

One thing I've noticed is that when editing the title or summary, hitting enter or escape either saves or discards the changes respectively (which is good), doesn't move the focus off of the input (which I found unexpected but may be good for keyboard navigation?), and then most unexpectedly in Brave, seems to make my cursor disappear. I can tell by typing more that the cursor is placed at the end of the input for either case, but it's invisible from that point out until I shift focus to another element and then back. It made for a somewhat odd user experience when testing, but I'm not quite sure what the right solution is at this moment.

A side consequence of making the title easier to change is that it's also now really easy to change the collection slug (which auto-updates to reflect the new title), and by extension, modify the public URL for the collection without realizing what you're doing. I could potentially see that causing issues for users in terms of links they've shared continuing to resolve. I wonder if we might want to change that so that slugs are not automatically updated on every title change in the same release that this goes out in?

SuaYoo reviewed (changes requested):
The new icons and truncation fixes look good. General feedback:
- It's not clear to me that the collection name and summary can edited in place since they appear the same as uneditable page names. Can we add back the pencil icon or make editable text visually distinct?
- Let's add a share option in the collection detail action menu to continue the pattern of representing all available buttons in the menu.
- I haven't tried this library out but it could be a good option for a tested text measure: https://github.com/chenglou/pretext/tree/main I found it while exploring alternatives for measuring tag lengths.

emma-sg committed:
update event format to match `BtrixChangeEvent`

emma-sg committed:
remove leftover comments

emma-sg committed:
remove redundant text measures

emma-sg committed:
add suffix slot + extra width for accommodating suffix icons

also adds those suffix icons

emma-sg committed:
blur input on ending editing

emma-sg committed:
add "share collection" to action menu

emma-sg commented:
@SuaYoo 

> It's not clear to me that the collection name and summary can edited in place since they appear the same as uneditable page names. Can we add back the pencil icon or make editable text visually distinct?

Good call, added as suffixes:

| Not editing | Editing
|--------|--------|
| <img width="343" height="90" alt="Screenshot 2026-05-12 at 7 55 50 PM" src="https://github.com/user-attachments/assets/a7b987d4-8a04-40b6-85fa-2e7124160076" /> | <img width="331" height="85" alt="Screenshot 2026-05-12 at 8 01 45 PM" src="https://github.com/user-attachments/assets/20b7231b-5a6a-40b6-8f06-32994d6f55d8" /> |


> Let's add a share option in the collection detail action menu to continue the pattern of representing all available buttons in the menu.

Sure, makes sense.

> I haven't tried this library out but it could be a good option for a tested text measure: [chenglou/pretext@main](https://github.com/chenglou/pretext/tree/main) I found it while exploring alternatives for measuring tag lengths.

Yes, pretext is very cool! It's a lot more than we need for measuring a single line, though – it has a whole layout engine for multiple lines of text. Take a look at how [`measure-text.ts`](https://github.com/webrecorder/browsertrix/pull/2943/changes#diff-e7e261c10c04a4c7678f6a7227baea646d5f80910c4142bf4fde805ecae54bc2) works though, it uses the same fast canvas-based text measurement API.

emma-sg commented:
@tw4l 

> One thing I've noticed is that when editing the title or summary, hitting enter or escape either saves or discards the changes respectively (which is good), doesn't move the focus off of the input (which I found unexpected but may be good for keyboard navigation?), and then most unexpectedly in Brave, seems to make my cursor disappear. I can tell by typing more that the cursor is placed at the end of the input for either case, but it's invisible from that point out until I shift focus to another element and then back. It made for a somewhat odd user experience when testing, but I'm not quite sure what the right solution is at this moment.

Oh, this was a bug caused by not defocusing the input element when leaving "edit" mode, fixed in b8dbaf4. Good catch, thank you!

> A side consequence of making the title easier to change is that it's also now really easy to change the collection slug (which auto-updates to reflect the new title), and by extension, modify the public URL for the collection without realizing what you're doing. I could potentially see that causing issues for users in terms of links they've shared continuing to resolve. I wonder if we might want to change that so that slugs are not automatically updated on every title change in the same release that this goes out in?

Yes, that's a good call, I think it'd make sense to expose it as an option in collection settings (or sharing settings maybe?). Will make an issue & PR for it!

tw4l commented:
When I save a name or summary by pressing enter while focused, I now see two toast messages pop up instead of one (this doesn't happen when saving by clicking outside the input with the mouse):

<img width="1216" height="689" alt="Screenshot 2026-05-13 at 10 57 11 AM" src="https://github.com/user-attachments/assets/cb72ba7e-b7d0-4f98-8656-8217742c36cf" />

emma-sg committed:
Update editable-text-field.ts

Co-authored-by: sua yoo <sua@webrecorder.org>

SuaYoo commented:
I believe `.renderContent` attribute was actually adding unnecessary re-renders that made the update width calls in `render()` work, even though it shouldn't since the `<span>` label doesn't exist on initial lifecycle calls. Since `<span>` inherits its font from the component, passing `this` to the text measurer seems to fix this. Changes: https://github.com/webrecorder/browsertrix/pull/3309/

SuaYoo committed:
Update collection editing and sharing UI - Rendering performance (#3309)

- Prevents `renderContent` from causing unnecessary re-renders
- Gets font from `<btrix-editable-text-field>` to remove dependency on
render call to get width of text

Moving from `renderContent` to slotted content should help prevent
similar issues, though this requires holding state in the parent.

emma-sg committed:
fix double-sent change events when pressing enter

the `save` fn wasn't idempotent, so it was sending two `btrix-change`
events once when called from the keyboard event handler, and again from
the input's `blur` handler

emma-sg commented:
@tw4l the double-toast issue is fixed in 18686c8, and @SuaYoo thank you for the perf/reactive update improvements!

tw4l reviewed (approved):
Thanks!
